presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+        if (!replace && existing.isPresent() && !ignoreExisting) {
+            throw new TableAlreadyExistsException(viewName);
+        }
+
+        List<ColumnMetadata> columns = new ArrayList<>();
+        for (ConnectorMaterializedViewDefinition.Column col : definition.getColumns()) {
+            columns.add(new ColumnMetadata(col.getName(), typeManager.getType(col.getType())));
+        }
+
+        Map<String, Object> finalProperties = new HashMap<>();
+        finalProperties.put(FILE_FORMAT_PROPERTY, FileFormat.PARQUET);
+        if (definition.getProperties().containsKey(PARTITIONING_PROPERTY)) {
+            finalProperties.put(PARTITIONING_PROPERTY, definition.getProperties().get(PARTITIONING_PROPERTY));
+        }
+
+        SchemaTableName storageTable = new SchemaTableName(viewName.getSchemaName(), storageTableName);
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(storageTable, columns, finalProperties, Optional.empty());
+
+        Optional<ConnectorNewTableLayout> layout = getNewTableLayout(session, tableMetadata);
+        finishCreateTable(session, beginCreateTable(session, tableMetadata, layout), ImmutableList.of(), ImmutableList.of());
+
+        tableBuilder.setViewOriginalText(Optional.of(encodeMaterializedViewData(definition)))
+                .setViewExpandedText(Optional.of("/* Presto Materialized View */"));
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+        if (existing.isPresent()) {
+            if (!replace) {
+                throw new MaterializedViewAlreadyExistsException(viewName);
+            }
+            // drop the current storage table
+            if (existing.get().getParameters().containsKey(STORAGE_TABLE)) {
+                String currentStorageTable = existing.get().getParameters().get(STORAGE_TABLE);
+                metastore.dropTable(identity, viewName.getSchemaName(), currentStorageTable, true);
+            }
+            metastore.replaceTable(identity, viewName.getSchemaName(), viewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        metastore.createTable(identity, table, principalPrivileges);
+    }
+
+    @Override
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        final HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> view = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+        if (!view.isPresent()) {
+            throw new MaterializedViewNotFoundException(viewName);
+        }
+
+        if (view.get().getParameters().containsKey(STORAGE_TABLE)) {
+            String storageTableName = view.get().getParameters().get(STORAGE_TABLE);
+            try {
+                metastore.dropTable(identity, viewName.getSchemaName(), storageTableName, true);
+            }
+            catch (Exception e) {
```

Comment:
We shouldn't catch arbitrary checked exceptions. This should probably just be `PrestoException`.

Why do we want to ignore errors when dropping the storage table? It seems like we should log a warning, at a minimum.
```java
catch (PrestoException e) {
    log.warn(e, "Failed to drop storage table '%s' for materialized view '%s', storageTableName, viewName);
}

Reply:
thanks. fixed.